### DISCUSSION TITLE
Add support for sudo

### DISCUSSION THREAD

**Discussion by aphsa** [umputun/spot#60] [Ideas]
Add support for sudo so that the scripts can be deployed as root or other users

**Comment by umputun** [comment]
Do you mean execute scripts on remote hosts with sudo? Or copy files to places not allowed to the ssh user? Technically, if this is about the first one, you can do sudo inside the script commands, but writing it multiple times for each line will get boring soon. So, adding the option "sudo: true" and running the script remotely with sudo makes sense to me.

However, using sudo for copy/sync/delete commands sounds strange. First of all, I have no clue how to do it with sftp client; probably, it requires some tricky copy to some /tmp, and after this, run `sudo cp` or smth like this. Another issue I'm having with this idea - if the goal is to operate on files that are outside of the rights of the given user, just use another ssh user or even root.

**Comment by umputun** [comment]
see https://github.com/umputun/spot/pull/62

**Reply by aphsa** [reply]
1. Correct. Upload as non-root and execute on target as sudo which Ansible does by creating /tmp/.ansible folders to copy the assets and later delete them once the playbook execution completes.
2. In most places I worked root login is disabled by security policy. Hence this request.

**Reply by umputun** [reply]
v0.14.0 includes sudo support for all the command types, except `sync`, see https://github.com/umputun/spot/pull/67
